psi4/driver/procrouting/proc.py

```diff
@@ -988,6 +989,46 @@ def select_mp4(name, **kwargs):
         return func(name, **kwargs)
 
 
+def select_adc2(name, **kwargs):
+    """Function selecting the algorithm for ADC(2) excited state energy
+    call and directing to specified or best-performance default modules.
+
+    """
+    reference = core.get_option('SCF', 'REFERENCE')
+    mtd_type = core.get_global_option('MP_TYPE')  # TODO or ADC_TYPE or CC_TYPE or ...
+    module = core.get_global_option('QC_MODULE')
+    # Considering only adcc/adc
+
+    # TODO Actually one should do selection on a couple of other options here
+    #      as well, e.g. adcc supports frozen-core and frozen-virtual,
+    #      spin-specific states or spin-flip methods.
```

Comment:
frozen-core as a selector is on my near-term todo list, so if you'd sketch in the logic, even in comments, that'd be great.

Reply:
Well, it's pretty simple: If I get it right adcc has frozen-core and frozen virtual, but the builtin has not, so one would go for adcc if it is available if any frozen core is desired.